src/utils/components/SSHSecretSection/hooks/useSecretsData.ts

```diff
@@ -4,45 +4,72 @@ import { modelToGroupVersionKind, SecretModel } from '@kubevirt-ui/kubevirt-api/
 import { IoK8sApiCoreV1Secret } from '@kubevirt-ui/kubevirt-api/kubernetes';
 import { ALL_NAMESPACES_SESSION_KEY } from '@kubevirt-utils/hooks/constants';
 import { isEmpty, validateSSHPublicKey } from '@kubevirt-utils/utils/utils';
-import { useK8sWatchResource, WatchK8sResult } from '@openshift-console/dynamic-plugin-sdk';
+import { useK8sWatchResources, WatchK8sResult } from '@openshift-console/dynamic-plugin-sdk';
 
 import { SecretSelectionOption, SSHSecretDetails } from '../utils/types';
 import { validateSecretNameLength, validateSecretNameUnique } from '../utils/utils';
 
 type UseSecretsData = (
   sshDetails: SSHSecretDetails,
-  namespace: string,
+  secretsNamespace: string,
+  currentNamespace: string,
 ) => {
   isDisabled: boolean;
   secretsData: WatchK8sResult<IoK8sApiCoreV1Secret[]>;
 };
 
-const useSecretsData: UseSecretsData = (sshDetails, namespace) => {
-  const [secrets, ...loadedAndErrorData] = useK8sWatchResource<IoK8sApiCoreV1Secret[]>({
-    groupVersionKind: modelToGroupVersionKind(SecretModel),
-    isList: true,
-    ...(namespace !== ALL_NAMESPACES_SESSION_KEY && {
-      namespace,
-    }),
-    limit: 10000,
-  });
+const useSecretsData: UseSecretsData = (sshDetails, secretsNamespace, currentNamespace) => {
```

Comment:
please check if secretsNamespace and currentNamespace arent the same

currentNamespace isn't activeNamespace? 

what happen if they are the same? u just fetching same thing twice

Reply:
`secretsNamespace` is the namespace selected in the "Use existing" section of the SSH Secret modal. `currentNamesace` is the namespace the VM is being created in or the namespace selected in the "Manage SSH keys" section on the settings page. 

They start off as the same and remain the same unless a different namespace is selected in the "Use existing" section.

I updated it to only pull the `secretsNamespace` secrets if `secretsNamespace` and `currentNamespace` are different.